EventFilter/HcalRawToDigi/plugins/PackerHelp.h

```diff
@@ -0,0 +1,464 @@
+     header |= (0x4 & TPHeaderSpec::MASK_FLAVOR)<<TPHeaderSpec::OFFSET_FLAVOER; //flavor
+     header |= (0x1 & TPHeaderSpec::MASK_HEADER_BIT)<<TPHeaderSpec::OFFSET_HEADER_BIT;
+
+     return header;
+  }
+
+  uhtrData* newUHTR( int uhtrIndex , int orn = 0 , int bcn = 0 , uint64_t evt = 0 ){
+    
+    // initialize vector of 16-bit words
+    uhtrs[uhtrIndex] = uhtrData(8);
+    // build header -- some information will be updated at the end    
+    
+    uint64_t presamples    = 10;
+    uint64_t uhtrCrate     = uhtrIndex&0xFF;
+    uint64_t uhtrSlot      = (uhtrIndex&0xF00)>>8; 
+// From Jeremy:
+// Set the firmware to zero, the firmware flavor to 0, the payload format to 1, and the event type to 1.
+    uint64_t fwFlavor      = 0;
+    uint64_t eventType     = 1;
+    uint64_t payloadFormat = 1;
+    uint64_t fwVersion     = 0;
+
+    uint64_t uhtrHeader1 = 0;
+    uhtrHeader1 |= (uint64_t(0x0)&MASK_DATA_LENGTH)<<OFFSET_DATA_LENGTH;
+    uhtrHeader1 |= (bcn&MASK_BCN)<<OFFSET_BCN;
+    uhtrHeader1 |= (evt&MASK_EVN)<<OFFSET_EVN;
+    uhtrHeader1 |= (uint64_t(0x0)&MASK_FILED_BY_AMC13)<<OFFSET_FILED_BY_AMC13;
+    
+    uint64_t uhtrHeader2 = 0;
+    uhtrHeader2 |= (uhtrCrate&MASK_CRATE_ID)<<OFFSET_CRATE_ID;
+    uhtrHeader2 |= (uhtrSlot&MASK_SLOT_ID)<<OFFSET_SLOT_ID;
+    uhtrHeader2 |= (presamples&MASK_PRESAMPLES)<<OFFSET_PRESAMPLES;
+    uhtrHeader2 |= (orn&MASK_ORN)<<OFFSET_ORN;
+    uhtrHeader2 |= (fwFlavor&MASK_FW_FLAVOR)<<OFFSET_FW_FLAVOR;
+    uhtrHeader2 |= (eventType&MASK_EVENT_TYPE)<<OFFSET_EVENT_TYPE;
+    uhtrHeader2 |= (payloadFormat&MASK_PAYLOAD_FORMAT)<<OFFSET_PAYLOAD_FORMAT;
+    uhtrHeader2 |= (fwVersion&MASK_FW_VERSION)<<OFFSET_FW_VERSION;
+
+    // push header into vector of 16-bit words
+    for (unsigned int i = 0; i< 4; ++i){
+       uhtrs[uhtrIndex][i] = ( uhtrHeader1>>(i*16) )&0xFFFF ;
+       uhtrs[uhtrIndex][i+4] = ( uhtrHeader2>>(i*16) )&0xFFFF ;
+    }
+
+    return &(uhtrs[uhtrIndex]);
+  };
+
+  void finalizeHeadTail(uhtrData* uhtr , bool verbosity){
+
+    uint64_t uhtr_size = uhtr->size()-8;
+
+    // adjust the size bits
+    uhtr->at(0) = uhtr_size&0xFFFF ;
+    uhtr->at(1) |= (uhtr_size>>16)&0xF ; 
+
+    unsigned int toAdd = 4-uhtr->size()%4;
+    for(unsigned int ia=0; ia<toAdd; ia++){
+       uhtr->push_back(0xD07F);
+    }
+
+    // add trailer
+    uhtr->push_back( uhtr_size&0xFFFF );
+    uhtr->push_back( (uhtr_size>>16)&0xF );
+    // this is ignoring the event number... I am not sure what this should be
+
+    // adding some blank stuff for the CRC bits
+    uhtr->push_back( 0 );
+    uhtr->push_back( 0 );
+  };
+
+  void addChannel( int uhtrIndex , edm::SortedCollection<HFDataFrame>::const_iterator& qiedf , const HcalElectronicsMap* readoutMap, int verbosity = 0 ){
+    if( qiedf->size() == 0 ) return;
+    DetId detid = qiedf->id();
+    HcalElectronicsId eid(readoutMap->lookup(detid));
+    uint16_t header = packQIE8header(qiedf->sample(0), eid);
```

Comment:
There is no accessor, to the best of my knowledge, in QIE10DataFrame and QIE11DataFrame for the electronics ID.  However, for HFDataFrame and HBHEDataFrames, its true that there are.  We can revert to those accessors if you think this would be best.